core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -431,6 +458,155 @@ object SparkSubmit {
   }
 }
 
+/** Provides utility functions to be used inside SparkSubmit. */
+private[spark] object SparkSubmitUtils extends Logging {
+
+  // Directories for caching downloads through ivy and storing the jars when maven coordinates are
+  // supplied to spark-submit
+  private var PACKAGES_DIRECTORY: File = null
```

Comment:
It looks like maybe this can be a local variable in `resolveMavenCoordinates` rather than a field in this object?